packages/sdk/src/modules/DatasourceUtility.ts

```diff
@@ -0,0 +1,48 @@
+import type { GetUnspentsResponse } from "../api/types";
+import type { Inscription } from "../inscription/types";
+import type { UTXO } from "../transactions/types";
+import { UNSTABLE_decodeObject } from "../utils";
+
+interface SegregateUTXOsBySpendStatusArgOptions {
+  utxos: UTXO[];
+}
+
+class DatasourceUtility {
+  static transformInscriptions(inscriptions?: Inscription[]) {
+    if (!inscriptions) return [];
+
+    return inscriptions.map((inscription) => ({
+      ...inscription,
+      meta: inscription.meta
+        ? UNSTABLE_decodeObject(inscription.meta)
+        : inscription.meta,
+    }));
+  }
```

Comment:
i think this function can just transform a single inscription like 
```ts
(inscription) => ({...inscription, meta: inscription.meta? ...})
```

then to use it for the RPC that returns multiple inscriptions just do `inscriptions.map(transformInscription)`

Reply:
We're using this function at 3 different places, all before reading decodeMetadata. I'll rename this function instead and accept the `decodeMetadata` parameter.

btw that `.map()` callback syntax is not advised! (see [Function callback risks](https://jakearchibald.com/2021/function-callback-risks/))